README.md

```diff
@@ -8,16 +8,21 @@ For a detailed explanation on why Glide doesn't do this see [here](http://engine
 
 ## Description
 
-This tool will help you removing from the project vendor directories all the files not needed for building your project. By default it'll keep all the files provided by packages returned by the `glide list` command.
-If you want to keep only go (including tests) files you can provide the `--only-code` option.
+This tool will help you removing from the project vendor directories all the files not needed for building your project. By default it'll keep only needed packages (returned by the `glide list` command) and all the files inside them.
+If you want to keep only source code (including tests) files you can provide the `--only-code` option.
 If you want to remove also the go test files you can add the `--no-tests` option.
 
 By default `glide-vc` doesn't remove:
 
 * files that are likely to contain some type of of legal declaration or licensing information (to remove them use the `--no-legal-files` option)
+* nested vendor directories. Doing this will change compilation and runtime behavior of your project because only the top level vendored dependencies will be used for compilation. If these are at a different revision (from the one provided inside nested vendor directories) they can cause compilation problems or runtime misbehiaviours. On the other side, keeping nested vendor directories can cause compilation problems like [this one](https://github.com/mattfarina/golang-broken-vendor).
 
-* nested vendor directories.
-Doing this will change compilation and runtime behavior of your project because only the top level vendored dependencies will be used for compilation. If these are at a different revision (from the one provided inside nested vendor directories) they can cause compilation problems or runtime misbehiaviours. On the other side, keeping nested vendor directories can cause compilation problems like [this one](https://github.com/mattfarina/golang-broken-vendor)
+## Use of glide.lock file instead of `glide list`
+By default `glide-vc` uses the output of the `glide list` command to retrieve the packages needed by your project. Currently, if you specify unneeded packages in your `glide.yaml` they will be installed and registred in the `glide.lock` file. So the `glide.lock` may contain also packages non required to build your project (but this behavior may change in future).
+
+Some projects use glide also for vendoring additional tools but `glide-vc` will remove them (since they aren't needed to build your project). Using the `--use-lock-file` option will make `glide-vc` use the packages list from `glide.lock` instead of the one provided by `glide list`, so it'll keep all the packages listed inside `glide.lock`.
```

Comment:
These paragraphs seem a tad disjointed. They seem to jump around between explanation and advice. Maybe they could be reordered as `[use case] -> [problem] -> [solution] -> [alternative]`?

> In order to vendor tools, some projects will specify packages in their `glide.yaml` that aren't imported by the project. These packages will be installed and registered in the `glide.lock` file (but this behavior may change in the future).
> 
> By default `glide-vc` uses the output of the `glide list` command to retrieve the packages needed by your project. Since `glide list` uses your projects imports it will not include the tools contained in `glide.lock`.
> 
> Using the `--use-lock-file` option will make `glide-vc` use the packages list from `glide.lock` instead of the one provided by `glide list`, preserving the tool packages.

And then the paragraph below as is